lib/internal/streams/writable.js

```diff
@@ -122,7 +122,13 @@ function WritableState(options, stream, isDuplex) {
   // Crypto is kind of old and crusty.  Historically, its default string
   // encoding is 'binary' so we have to make this configurable.
   // Everything else in the universe uses 'utf8', though.
-  this.defaultEncoding = (options && options.defaultEncoding) || 'utf8';
+  const defaultEncoding = options?.defaultEncoding;
+
+  if (defaultEncoding && !Buffer.isEncoding(defaultEncoding)) {
+    throw new ERR_UNKNOWN_ENCODING(defaultEncoding);
+  }
+
+  this.defaultEncoding = defaultEncoding || 'utf8';
```

Comment:
For falsy values you are omitting throwing error, such as null and false.

Reply:
I left the same behaviour as before when passing null, going into default

Reply:
no need to throw, its converted to `utf8` as it was before my fix